```
 Formatting [1256/1862] librz/include/rz_cmd.h
librz/include/rz_cmd.h:560:50: error: code should be clang-formatted [-Wclang-format-violations]
#define rz_cmd_desc_children_foreach(root, it_cd) rz_pvector_foreach (&root->children, it_cd)
                                                 ^
```

--- a/librz/core/cmd/cmd_api.c
+++ b/librz/core/cmd/cmd_api.c
@@ -32,8 +32,21 @@
 #define MAX_RIGHT_ALIGHNMENT 20
 
 // Based on Config and Line order choses the vertical line shape
-static inline const char *get_vertical_line(int idx, int len, bool scr_curvy, bool scr_utf8) {
-	return (((idx == 0) && (scr_curvy == true)) ? RUNE_CURVE_CORNER_TL : (((idx == len - 1) && (scr_curvy == true)) ? RUNE_CURVE_CORNER_BL : ((scr_utf8 == true) ? RUNE_LINE_VERT : "| ")));
+static inline const char *get_vertical_line(size_t idx, size_t len, bool scr_curvy, bool scr_utf8) {
+	if (scr_curvy && !idx) {
+		return RUNE_CURVE_CORNER_TL;
+	} else if (scr_curvy && (idx == len - 1)) {
+		return RUNE_CURVE_CORNER_BL;
+	} else if (scr_utf8) {
+		return RUNE_LINE_VERT;
+	} else {
+		return "| ";
+	}
+}
+
+// return true if core and core->config aren't NULL, else false
+static inline bool is_core_and_config(RzCore *core) {
+	return (core && core->config);
 }
 
 // NOTE: this should be in sync with SPECIAL_CHARACTERS in
@@ -1120,14 +1133,18 @@ static void print_child_help(RzCmd *cmd, RzStrBuf *sb, RzCmdDesc *cd, size_t max
 }
 
 static char *group_get_help(RzCmd *cmd, RzCmdDesc *cd, bool use_color) {
-	bool scr_utf8 = cmd->core ? cmd->core->config ? rz_config_get_b(cmd->core->config, "scr.utf8") : false : false;
-	bool scr_curvy = scr_utf8 && rz_config_get_b(cmd->core->config, "scr.utf8.curvy");
+	bool scr_utf8 = false, scr_curvy = false;
+	if (is_core_and_config(cmd->core)) {
+		scr_utf8 = rz_config_get_b(cmd->core->config, "scr.utf8");
+		scr_curvy = scr_utf8 && rz_config_get_b(cmd->core->config, "scr.utf8.curvy");
+	}
 
 	RzStrBuf *sb = rz_strbuf_new(NULL);
 	fill_usage_strbuf(cmd, sb, cd, use_color);
 
 	void **it_cd;
-	int idx;
+	size_t idx = 0;
+
 	size_t max_len = 0, min_len = SIZE_MAX;
 
 	rz_cmd_desc_children_foreach(cd, it_cd) {
@@ -1138,9 +1155,12 @@ static char *group_get_help(RzCmd *cmd, RzCmdDesc *cd, bool use_color) {
 		max_len = min_len + MAX_RIGHT_ALIGHNMENT;
 	}
 
+	size_t n_descs = rz_pvector_len(&cd->children);
+
 	rz_cmd_desc_children_foreach_idx(cd, it_cd, idx) {
 		RzCmdDesc *child = *(RzCmdDesc **)it_cd;
-		print_child_help(cmd, sb, child, max_len, get_vertical_line(idx, cd->children.v.len, scr_curvy, scr_utf8), use_color);
+		const char *vertical_line = get_vertical_line(idx, n_descs, scr_curvy, scr_utf8);
+		print_child_help(cmd, sb, child, max_len, vertical_line, use_color);
 	}
 
 	bool details_filled = fill_details(cmd, cd, sb, use_color);
@@ -1173,19 +1193,25 @@ static char *group_get_help(RzCmd *cmd, RzCmdDesc *cd, bool use_color) {
 }
 
 static void fill_argv_modes_help_strbuf(RzCmd *cmd, RzStrBuf *sb, RzCmdDesc *cd, bool use_color) {
+	bool scr_utf8 = false;
+	if (is_core_and_config(cmd->core)) {
+		scr_utf8 = rz_config_get_b(cmd->core->config, "scr.utf8");
+	}
+	const char *vertical_line = scr_utf8 ? RUNE_LINE_VERT : "| ";
+
 	size_t max_len = 0, min_len = SIZE_MAX;
 	update_minmax_len(cd, &max_len, &min_len, true);
 	max_len++; // consider the suffix letter
 	if (max_len - min_len > MAX_RIGHT_ALIGHNMENT) {
 		max_len = min_len + MAX_RIGHT_ALIGHNMENT;
 	}
-	bool scr_utf8 = cmd->core ? cmd->core->config ? rz_config_get_b(cmd->core->config, "scr.utf8") : false : false;
 	size_t i;
+
 	for (i = 0; i < RZ_ARRAY_SIZE(argv_modes); i++) {
 		if (cd->d.argv_modes_data.modes & argv_modes[i].mode) {
 			char *name = rz_str_newf("%s%s", cd->name, argv_modes[i].suffix);
 			char *summary = rz_str_newf("%s%s", cd->help->summary, argv_modes[i].summary_suffix);
-			do_print_child_help(cmd, sb, cd, name, summary, scr_utf8 ? RUNE_LINE_VERT : "| ", false, max_len, use_color);
+			do_print_child_help(cmd, sb, cd, name, summary, vertical_line, false, max_len, use_color);
 			free(name);
 			free(summary);
 		}
@@ -1492,8 +1518,14 @@ RZ_API bool rz_cmd_get_help_json(RzCmd *cmd, const RzCmdDesc *cd, PJ *j) {
  */
 RZ_API bool rz_cmd_get_help_strbuf(RzCmd *cmd, const RzCmdDesc *cd, bool use_color, RzStrBuf *sb) {
 	rz_return_val_if_fail(cmd && cd && sb, false);
-	bool scr_utf8 = cmd->core ? cmd->core->config ? rz_config_get_b(cmd->core->config, "scr.utf8") : false : false;
-	do_print_child_help(cmd, sb, cd, cd->name, cd->help->summary, scr_utf8 ? RUNE_LINE_VERT : "| ", false, MAX_RIGHT_ALIGHNMENT, use_color);
+
+	bool scr_utf8 = false;
+	if (is_core_and_config(cmd->core)) {
+		scr_utf8 = rz_config_get_b(cmd->core->config, "scr.utf8");
+	}
+	const char *vertical_line = scr_utf8 ? RUNE_LINE_VERT : "| ";
+
+	do_print_child_help(cmd, sb, cd, cd->name, cd->help->summary, vertical_line, false, MAX_RIGHT_ALIGHNMENT, use_color);
 	return true;
 }
 

--- a/librz/core/cmd/cmd_api.c
+++ b/librz/core/cmd/cmd_api.c
@@ -39,14 +39,13 @@ static inline const char *get_vertical_line(size_t idx, size_t len, bool scr_cur
 		return RUNE_CURVE_CORNER_BL;
 	} else if (scr_utf8) {
 		return RUNE_LINE_VERT;
-	} else {
-		return "| ";
 	}
+	return "| ";
 }
 
-// return true if core and core->config aren't NULL, else false
-static inline bool is_core_and_config(RzCore *core) {
-	return (core && core->config);
+// return rz_config_get_b if core and core->config aren't NULL else false
+static inline bool core_config_get_b(RzCore *core, const char *option) {
+	return core && core->config ? rz_config_get_b(core->config, option) : false;
 }
 
 // NOTE: this should be in sync with SPECIAL_CHARACTERS in
@@ -1133,11 +1132,8 @@ static void print_child_help(RzCmd *cmd, RzStrBuf *sb, RzCmdDesc *cd, size_t max
 }
 
 static char *group_get_help(RzCmd *cmd, RzCmdDesc *cd, bool use_color) {
-	bool scr_utf8 = false, scr_curvy = false;
-	if (is_core_and_config(cmd->core)) {
-		scr_utf8 = rz_config_get_b(cmd->core->config, "scr.utf8");
-		scr_curvy = scr_utf8 && rz_config_get_b(cmd->core->config, "scr.utf8.curvy");
-	}
+	bool scr_utf8 = core_config_get_b(cmd->core, "scr.utf8");
+	bool scr_curvy = scr_utf8 && core_config_get_b(cmd->core, "scr.utf8.curvy");
 
 	RzStrBuf *sb = rz_strbuf_new(NULL);
 	fill_usage_strbuf(cmd, sb, cd, use_color);
@@ -1193,10 +1189,7 @@ static char *group_get_help(RzCmd *cmd, RzCmdDesc *cd, bool use_color) {
 }
 
 static void fill_argv_modes_help_strbuf(RzCmd *cmd, RzStrBuf *sb, RzCmdDesc *cd, bool use_color) {
-	bool scr_utf8 = false;
-	if (is_core_and_config(cmd->core)) {
-		scr_utf8 = rz_config_get_b(cmd->core->config, "scr.utf8");
-	}
+	bool scr_utf8 = core_config_get_b(cmd->core, "scr.utf8");
 	const char *vertical_line = scr_utf8 ? RUNE_LINE_VERT : "| ";
 
 	size_t max_len = 0, min_len = SIZE_MAX;
@@ -1519,10 +1512,7 @@ RZ_API bool rz_cmd_get_help_json(RzCmd *cmd, const RzCmdDesc *cd, PJ *j) {
 RZ_API bool rz_cmd_get_help_strbuf(RzCmd *cmd, const RzCmdDesc *cd, bool use_color, RzStrBuf *sb) {
 	rz_return_val_if_fail(cmd && cd && sb, false);
 
-	bool scr_utf8 = false;
-	if (is_core_and_config(cmd->core)) {
-		scr_utf8 = rz_config_get_b(cmd->core->config, "scr.utf8");
-	}
+	bool scr_utf8 = core_config_get_b(cmd->core, "scr.utf8");
 	const char *vertical_line = scr_utf8 ? RUNE_LINE_VERT : "| ";
 
 	do_print_child_help(cmd, sb, cd, cd->name, cd->help->summary, vertical_line, false, MAX_RIGHT_ALIGHNMENT, use_color);

--- a/librz/include/rz_cmd.h
+++ b/librz/include/rz_cmd.h
@@ -557,7 +557,7 @@ RZ_API bool rz_cmd_desc_remove(RzCmd *cmd, RzCmdDesc *cd);
 RZ_API void rz_cmd_foreach_cmdname(RzCmd *cmd, RzCmdDesc *begin, RzCmdForeachNameCb cb, void *user);
 RZ_API const RzCmdDescArg *rz_cmd_desc_get_arg(const RzCmdDesc *cd, size_t i);
 
-#define rz_cmd_desc_children_foreach(root, it_cd) rz_pvector_foreach (&root->children, it_cd)
+#define rz_cmd_desc_children_foreach(root, it_cd)          rz_pvector_foreach (&root->children, it_cd)
 #define rz_cmd_desc_children_foreach_idx(root, it_cd, idx) rz_pvector_enumerate (&root->children, it_cd, idx)
 
 RZ_API void rz_cmd_desc_details_free(RzCmdDescDetail *details);

